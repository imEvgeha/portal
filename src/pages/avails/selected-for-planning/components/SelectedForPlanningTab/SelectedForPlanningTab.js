import React, {useEffect, useState} from 'react';
import PropTypes from 'prop-types';
import {URL} from '../../../../../util/Common';
import {SELECTED_FOR_PLANNING_TAB} from '../../constants';
import NexusTooltip from '../../../../../ui/elements/nexus-tooltip/NexusTooltip';
import DOPService from '../../DOP-services';

const SelectedForPlanningTab = ({activeTab, setActiveTab}) => {
    const [rightsCount, setRightsCount] = useState(0);

    //TODO: Resolve DOP issue
   /* useEffect(() => {
        DOPService.getUsersProjectsList(1,1).then( res => {
            setRightsCount(1);
        });
    }, [activeTab]);*/

    const setPlanning = () => {
        setActiveTab(SELECTED_FOR_PLANNING_TAB);
    };

    return (
        URL.isLocalOrDevOrQA() && (
            <NexusTooltip content='Click to refresh' isDisabled={activeTab !== SELECTED_FOR_PLANNING_TAB}>
                <div
                    className={`nexus-c-table-toolbar__title 
                        ${activeTab === SELECTED_FOR_PLANNING_TAB ? 'nexus-c-table-toolbar__title--is-active' : ''}`}
                    onClick={setPlanning}
                >
                    {SELECTED_FOR_PLANNING_TAB} ({rightsCount})
                </div>
            </NexusTooltip>
        )
    );
};

SelectedForPlanningTab.propTypes = {
    activeTab: PropTypes.bool.isRequired,
    setActiveTab: PropTypes.func.isRequired,
};

SelectedForPlanningTab.defaultProps = {};

export default SelectedForPlanningTab;