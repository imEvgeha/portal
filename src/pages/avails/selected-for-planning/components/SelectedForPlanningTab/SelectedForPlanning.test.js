import React from 'react';
import {shallow} from 'enzyme';
import SelectedForPlanningTab from './SelectedForPlanningTab';

describe('SelectedForPlanningTab', () => {
    let wrapper = null;
    const props = {};
    beforeAll(() => {
        wrapper = shallow(<SelectedForPlanningTab {...props} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});