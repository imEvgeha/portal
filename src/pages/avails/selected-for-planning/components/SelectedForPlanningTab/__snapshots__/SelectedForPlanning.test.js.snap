// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedForPlanningTab should match snapshot 1`] = `
<NexusTooltip
  content="Click to refresh"
  isDisabled={true}
>
  <div
    className="nexus-c-table-toolbar__title 
                        "
    onClick={[Function]}
  >
    Selected for Planning
     (
    0
    )
  </div>
</NexusTooltip>
`;
